apps/playground/src/app/modules/components/checkbox-module/checkbox-module.component.ts

```diff
@@ -66,72 +34,32 @@ export class CheckboxModuleComponent implements OnInit {
     }
   ];
 
-  successItems = [
-    {
-      label: '1',
-      id: 'success_1',
-      value: '1',
-      name: '1'
-    },
-    {
-      label: '2',
-      id: 'success_2',
-      value: '2',
-      name: '2'
-    }
-  ];
-
-  errorItems = [
-    {
-      label: '1',
-      id: 'error_1',
-      value: '1',
-      name: '1'
-    },
-    {
-      label: '2',
-      id: 'error_2',
-      value: '2',
-      name: '2'
-    }
-  ];
-
-  disabledItems = [
-    {
-      label: '1',
-      id: 'disabled_1',
-      value: '1',
-      name: '1'
-    },
-    {
-      label: '2',
-      id: 'disabled_2',
-      value: '2',
-      name: '2'
-    }
-  ];
+  possibleSizes: Record<string, string> = {
+    'sm': 'sm',
+    'md': 'md'
+  };
 
-  templateDrivenSelected = ['2'];
-  reactiveFormsGroup: FormGroup | undefined;
-  inlineModel = ['1'];
-  successModel = ['1'];
-  errorModel = ['1'];
-  disabledModel = ['1'];
-  smallSize = ElementSize.SMALL;
-
-  constructor(private formBuilder: FormBuilder) { }
+  constructor( private formBuilder: NonNullableFormBuilder ) {
+  }
 
   ngOnInit(): void {
     this.initForms();
   }
 
   private initForms(): void {
-    this.form = this.formBuilder.group({
-      label: 'DSFR checkbox works 😁'
-    });
-
-    this.reactiveFormsGroup = this.formBuilder.group({
-      checkbox: [['1']]
-    });
+    this.checkboxForm = this.formBuilder.group( {
+      legend: new FormControl( 'This is a legend', [ Validators.required ] ),
+      hint: new FormControl( 'This is a hint', [ Validators.required ] ),
+      inline: new FormControl( false, [ Validators.required ] ),
+      disabled: new FormControl( false, [ Validators.required ] ),
+      size: new FormControl( ElementSize.MEDIUM, [ Validators.required ] ),
+      hasFailed: new FormControl( false, [ Validators.required ] ),
+      failureMessage: new FormControl( 'This is a failure message', [ Validators.required ] ),
+      hasSucceeded: new FormControl( false, [ Validators.required ] ),
+      successMessage: new FormControl( 'This is a success message', [ Validators.required ] )
+    } );
+    this.form = this.formBuilder.group( {
+      checkbox: new FormControl( [] )
+    } );
```

Comment:
If we want the `NonNullableFormBuilder` to work, we apparently need the other syntax : 
```typescript
this.checkboxForm = this.formBuilder.group({
      legend: ['This is a legend', [Validators.required]],
      hint: ['This is a hint', [Validators.required]],
      inline: [false, [Validators.required]],
      disabled: [false, [Validators.required]],
      size: [ElementSize.MEDIUM, [Validators.required]],
      hasFailed: [false, [Validators.required]],
      failureMessage: ['This is a failure message', [Validators.required]],
      hasSucceeded: [false, [Validators.required]],
      successMessage: ['This is a success message', [Validators.required]]
    });
    this.form = this.formBuilder.group({
      checkbox: [['']]
    });
```